homeassistant/components/soundtouch/media_player.py

```diff
@@ -236,7 +263,11 @@ def is_volume_muted(self):
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        return SUPPORT_SOUNDTOUCH
+        return (
+            (SUPPORT_SOUNDTOUCH | SUPPORT_SELECT_SOURCE)
+            if len(self._sources) > 0
+            else SUPPORT_SOUNDTOUCH
```

Comment:
Please don't use a ternary operator if it'll wrap multiple lines.

Make a local variable that we populate with the correct features and return the variable at the end.

Reply:
done.